benches/distribution_statistic.rs

```diff
@@ -0,0 +1,73 @@
+use criterion::{criterion_group, criterion_main, BatchSize, Criterion};
+
+use rand::distributions::Distribution;
+use rand::distributions::Uniform;
+use rand::rngs::SmallRng;
+use rand::seq::SliceRandom;
+use rand::SeedableRng;
+
+use vector::event::metric::Sample;
+use vector::sinks::util::statistic::DistributionStatistic;
+
+fn generate_samples(mut size: u32, max_bin_count: u32) -> Vec<Sample> {
+    // generate random samples, but we also want to use
+    // the same samples set on each run.
+    let mut rng = SmallRng::seed_from_u64(1234);
+    let range = Uniform::from(1..=max_bin_count);
+    let mut value = 1.0;
+    let mut samples = Vec::new();
+    while size > 0 {
+        let bin_count = u32::min(range.sample(&mut rng), size);
```

Comment:
This will create smaller bin counts each time through the loop, leading to a very uneven distribution of values, heavily weighted towards the first samples.

Reply:
Only the last bin is affected by `min` call, I think all other bins should be uniformly distributed in [1..range].

Reply:
Oops, I misread that as generating random numbers based only on `size`, which shrinks each time through the loop. My bad.